nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -71,15 +73,9 @@ public static string[] GetTargetFrameworkMonikers(ImmutableArray<ProjectBuildFil
         foreach (var targetFrameworkValue in targetFrameworkValues)
         {
             var tfms = targetFrameworkValue;
-            if (tfms.StartsWith("$(") && tfms.EndsWith(")"))
+            if (HasPropertyReplace(tfms))
             {
-                var propertyName = tfms.Substring(2, tfms.Length - 3);
-                while (propertyName is not null)
-                {
-                    propertyName = propertyInfo.TryGetValue(propertyName, out tfms) && tfms.StartsWith("$(") && tfms.EndsWith(")")
-                        ? tfms.Substring(2, tfms.Length - 3)
-                        : null;
-                }
+                tfms = GetRootValue(tfms, propertyInfo);
```

Comment:
Could always call `GetRoolValue` here. Then all use of `HasPropertyReplace` and `GetPropertyName` will be done in pairs making a good case for them to become a single `TryGetPropertyName` method.

Reply:
Love that, good catch.